ui/app/src/components/CreateFileDialog/CreateFileDialogContainer.tsx

```diff
@@ -51,7 +51,10 @@ export function CreateFileDialogContainer(props: CreateFileContainerProps) {
   const { formatMessage } = useIntl();
   const itemLookup = useItemsByPath();
   const computedFilePath = `${path}/${getFileNameWithExtensionForItemType(type, name)}`;
-  const fileExists = itemLookup[computedFilePath] !== UNDEFINED;
+  // When calling the validation API, we need to check if the item with the suggested name exists. This is an extra validation for the
+  // fileExists const.
+  const [itemExists, setItemExists] = useState(false);
+  const fileExists = itemLookup[computedFilePath] !== UNDEFINED || itemExists;
   const isValid = !isBlank(name) && !fileExists;
 
   const onCreateFile = (site: string, path: string, fileName: string) => {
```

Comment:
Looks like this function checks for existence too. Perhaps we limit this one to create since the other one is doing all the checking for existence? Why is this function separate; is it used somewhere else? Check to see if we can optimize and consolidate.

Reply:
onCreateFile is called in a couple of places, when validating the site policy there may be a name suggestion, if there is a suggestion we display the confirmation then we create the file (if suggestion accepted). If validation is successful, we directly create the file. 
Though the existence check can be removed in `onCreateFile` since those validations are already in `onCreate`. 
Also, I renamed `onCreate` to `onSubmit` since I feel is more accurate.